CI fails for distrbuted test_rloo[fsdp2]: CUDA error: device-side assert triggered

CI fails for distrbuted test_rloo[fsdp2]: https://github.com/huggingface/trl/actions/runs/23628572685/job/68822753991
> torch.AcceleratorError: CUDA error: device-side assert triggered
```python
FAILED tests/distributed/test_distributed.py::TestDistributed::test_rloo[fsdp2] - AssertionError: assert 1 == 0
 +  where 1 = CompletedProcess(args=['accelerate', 'launch', '--config_file', PosixPath('/tmp/pytest-of-root/pytest-0/test_rloo_fsdp2_0/data/accelerate_configs/fsdp2.yaml'), 'trl/scripts/rloo.py', '--output_dir', '/tmp/pytest-of-root/pytest-0/test_rloo_fsdp2_0', '--model_name_or_path', 'trl-internal-testing/tiny-Qwen2ForCausalLM-2.5', '--dataset_name', 'trl-internal-testing/zen', '--dataset_config', 'conversational_prompt_only', '--reward_model_name_or_path', 'trl-internal-testing/tiny-Qwen2ForSequenceClassification-2.5'], returncode=1).returncode
```

Stacktrace:
```python
_______________________ TestDistributed.test_rloo[fsdp2] _______________________

self = <tests.distributed.test_distributed.TestDistributed object at 0x7ff53a844ad0>
config = 'fsdp2'
get_config_path = <function get_config_path.<locals>._get_config_path at 0x7ff53a50fd80>

    @pytest.mark.parametrize(
        "config",
        [
            "ddp",
            pytest.param(
                "zero2",
                marks=pytest.mark.xfail(
                    Version(transformers.__version__) == Version("5.1.0"),
                    reason="Upstream incompatibility: deepspeed and transformers==5.1.0 (see transformers#43780)",
                ),
            ),
            pytest.param("zero3", marks=pytest.mark.xfail(reason="ZeRO 3 is currently failing, see #4899")),
            "fsdp2",
        ],
    )
    def test_rloo(self, config, get_config_path):
        # fmt: off
>       run_command(
            [
                "accelerate", "launch", "--config_file", get_config_path(config), "trl/scripts/rloo.py",
                "--output_dir", self.tmp_dir,
                "--model_name_or_path", "trl-internal-testing/tiny-Qwen2ForCausalLM-2.5",
                "--dataset_name", "trl-internal-testing/zen",
                "--dataset_config", "conversational_prompt_only",
                "--reward_model_name_or_path", "trl-internal-testing/tiny-Qwen2ForSequenceClassification-2.5",
            ],
            os.environ.copy(),
        )

tests/distributed/test_distributed.py:242: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['accelerate', 'launch', '--config_file', PosixPath('/tmp/pytest-of-root/pytest-0/test_rloo_fsdp2_0/data/accelerate_configs/fsdp2.yaml'), 'trl/scripts/rloo.py', '--output_dir', ...]
env = {'ACTIONS_ORCHESTRATION_ID': '34e8d70d-b149-49b6-baf2-06c01e941646.distributed_smoke.__default', 'CI': 'true', 'CI_SLACK_CHANNEL': '***', 'CUDA_VERSION': '12.8.1', ...}

    def run_command(command: list[str], env: dict[str, str]) -> None:
        result = subprocess.run(command, env=env, cwd=ROOT)
>       assert result.returncode == 0
E       AssertionError: assert 1 == 0
E        +  where 1 = CompletedProcess(args=['accelerate', 'launch', '--config_file', PosixPath('/tmp/pytest-of-root/pytest-0/test_rloo_fsdp2_0/data/accelerate_configs/fsdp2.yaml'), 'trl/scripts/rloo.py', '--output_dir', '/tmp/pytest-of-root/pytest-0/test_rloo_fsdp2_0', '--model_name_or_path', 'trl-internal-testing/tiny-Qwen2ForCausalLM-2.5', '--dataset_name', 'trl-internal-testing/zen', '--dataset_config', 'conversational_prompt_only', '--reward_model_name_or_path', 'trl-internal-testing/tiny-Qwen2ForSequenceClassification-2.5'], returncode=1).returncode

tests/distributed/test_distributed.py:32: AssertionError
----------------------------- Captured stderr call -----------------------------
Warning: You are sending unauthenticated requests to the HF Hub. Please set a HF_TOKEN to enable higher rate limits and faster downloads.
Warning: You are sending unauthenticated requests to the HF Hub. Please set a HF_TOKEN to enable higher rate limits and faster downloads.
The tokenizer has new PAD/BOS/EOS tokens that differ from the model config and generation config. The model config and generation config were aligned accordingly, being updated with the tokenizer's values. Updated tokens: {'eos_token_id': 151645, 'bos_token_id': None, 'pad_token_id': 151643}.
The tokenizer has new PAD/BOS/EOS tokens that differ from the model config and generation config. The model config and generation config were aligned accordingly, being updated with the tokenizer's values. Updated tokens: {'eos_token_id': 151645, 'bos_token_id': None, 'pad_token_id': 151643}.
Passing `generation_config` together with generation-related arguments=({'disable_compile'}) is deprecated and will be removed in future versions. Please pass either a `generation_config` object OR all generation parameters explicitly, but not both.
Passing `generation_config` together with generation-related arguments=({'disable_compile'}) is deprecated and will be removed in future versions. Please pass either a `generation_config` object OR all generation parameters explicitly, but not both.
/pytorch/aten/src/ATen/native/cuda/TensorCompare.cu:109: _assert_async_cuda_kernel: block: [0,0,0], thread: [0,0,0] Assertion `probability tensor contains either `inf`, `nan` or element < 0` failed.
/pytorch/aten/src/ATen/native/cuda/TensorCompare.cu:109: _assert_async_cuda_kernel: block: [0,0,0], thread: [0,0,0] Assertion `probability tensor contains either `inf`, `nan` or element < 0` failed.
[rank1]: Traceback (most recent call last):
[rank1]:   File "/__w/trl/trl/trl/scripts/rloo.py", line 169, in <module>
[rank1]:     main(script_args, training_args, model_args, dataset_args)
[rank1]:   File "/__w/trl/trl/trl/scripts/rloo.py", line 141, in main
[rank1]:     trainer.train()
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/trainer.py", line 1424, in train
[rank1]:     return inner_training_loop(
[rank1]:            ^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/trainer.py", line 1506, in _inner_training_loop
[rank1]:     self._run_epoch(
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/trainer.py", line 1734, in _run_epoch
[rank1]:     tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
[rank1]:                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/trainer/rloo_trainer.py", line 748, in training_step
[rank1]:     output = super().training_step(model, inputs, num_items_in_batch)
[rank1]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/trainer.py", line 1900, in training_step
[rank1]:     inputs = self._prepare_inputs(inputs)
[rank1]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/extras/profiling.py", line 202, in wrapper
[rank1]:     return func(self, *args, **kwargs)
[rank1]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/trainer/rloo_trainer.py", line 777, in _prepare_inputs
[rank1]:     generation_batch = self._generate_and_score_completions(generation_batch)
[rank1]:                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/trainer/rloo_trainer.py", line 1118, in _generate_and_score_completions
[rank1]:     prompt_ids_list, completion_ids_list, completions = self._generate(prompts)
[rank1]:                                                         ^^^^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/trainer/rloo_trainer.py", line 1043, in _generate
[rank1]:     completion_ids = self._generate_single_turn(prompt_ids, images, multimodal_fields)
[rank1]:                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/trainer/rloo_trainer.py", line 1016, in _generate_single_turn
[rank1]:     prompt_completion_ids = unwrapped_model.generate(
[rank1]:                             ^^^^^^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/torch/distributed/fsdp/_fully_shard/_fully_shard.py", line 689, in wrapped_method
[rank1]:     out = orig_method(*args, **kwargs)
[rank1]:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/torch/utils/_contextlib.py", line 124, in decorate_context
[rank1]:     return func(*args, **kwargs)
[rank1]:            ^^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/generation/utils.py", line 2521, in generate
[rank1]:     result = decoding_method(
[rank1]:              ^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/generation/utils.py", line 2721, in _sample
[rank1]:     while self._has_unfinished_sequences(this_peer_finished, synced_gpus, device=input_ids.device):
[rank1]:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank1]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/generation/utils.py", line 2541, in _has_unfinished_sequences
[rank1]:     this_peer_finished_flag = torch.tensor(0.0 if this_peer_finished else 1.0, device=device)
[rank1]:                                                   ^^^^^^^^^^^^^^^^^^
[rank1]: torch.AcceleratorError: CUDA error: device-side assert triggered
[rank1]: Search for `cudaErrorAssert' in https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html for more information.
[rank1]: CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
[rank1]: For debugging consider passing CUDA_LAUNCH_BLOCKING=1
[rank1]: Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

[rank0]: Traceback (most recent call last):
[rank0]:   File "/__w/trl/trl/trl/scripts/rloo.py", line 169, in <module>
[rank0]:     main(script_args, training_args, model_args, dataset_args)
[rank0]:   File "/__w/trl/trl/trl/scripts/rloo.py", line 141, in main
[rank0]:     trainer.train()
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/trainer.py", line 1424, in train
[rank0]:     return inner_training_loop(
[rank0]:            ^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/trainer.py", line 1506, in _inner_training_loop
[rank0]:     self._run_epoch(
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/trainer.py", line 1734, in _run_epoch
[rank0]:     tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
[rank0]:                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/trainer/rloo_trainer.py", line 748, in training_step
[rank0]:     output = super().training_step(model, inputs, num_items_in_batch)
[rank0]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/trainer.py", line 1900, in training_step
[rank0]:     inputs = self._prepare_inputs(inputs)
[rank0]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/extras/profiling.py", line 202, in wrapper
[rank0]:     return func(self, *args, **kwargs)
[rank0]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/trainer/rloo_trainer.py", line 777, in _prepare_inputs
[rank0]:     generation_batch = self._generate_and_score_completions(generation_batch)
[rank0]:                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/trainer/rloo_trainer.py", line 1118, in _generate_and_score_completions
[rank0]:     prompt_ids_list, completion_ids_list, completions = self._generate(prompts)
[rank0]:                                                         ^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/trainer/rloo_trainer.py", line 1043, in _generate
[rank0]:     completion_ids = self._generate_single_turn(prompt_ids, images, multimodal_fields)
[rank0]:                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/trl/trainer/rloo_trainer.py", line 1016, in _generate_single_turn
[rank0]:     prompt_completion_ids = unwrapped_model.generate(
[rank0]:                             ^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/torch/distributed/fsdp/_fully_shard/_fully_shard.py", line 689, in wrapped_method
[rank0]:     out = orig_method(*args, **kwargs)
[rank0]:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/torch/utils/_contextlib.py", line 124, in decorate_context
[rank0]:     return func(*args, **kwargs)
[rank0]:            ^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/generation/utils.py", line 2521, in generate
[rank0]:     result = decoding_method(
[rank0]:              ^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/generation/utils.py", line 2721, in _sample
[rank0]:     while self._has_unfinished_sequences(this_peer_finished, synced_gpus, device=input_ids.device):
[rank0]:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank0]:   File "/__w/trl/trl/.venv/lib/python3.12/site-packages/transformers/generation/utils.py", line 2541, in _has_unfinished_sequences
[rank0]:     this_peer_finished_flag = torch.tensor(0.0 if this_peer_finished else 1.0, device=device)
[rank0]:                                                   ^^^^^^^^^^^^^^^^^^
[rank0]: torch.AcceleratorError: CUDA error: device-side assert triggered
[rank0]: Search for `cudaErrorAssert' in https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html for more information.
[rank0]: CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
[rank0]: For debugging consider passing CUDA_LAUNCH_BLOCKING=1
[rank0]: Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

[rank1]:[E327 02:53:46.528543289 ProcessGroupNCCL.cpp:2119] [PG ID 0 PG GUID 0(default_pg) Rank 1] Process group watchdog thread terminated with exception: CUDA error: device-side assert triggered
Search for `cudaErrorAssert' in https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html for more information.
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

Exception raised from query at /pytorch/c10/cuda/CUDAEvent.h:111 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x9d (0x7f1204d4e05d in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libc10.so)
frame #1: <unknown function> + 0xc114 (0x7f1204de7114 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libc10_cuda.so)
frame #2: c10d::ProcessGroupNCCL::WorkNCCL::finishedGPUExecutionInternal() const + 0x50 (0x7f11502f3d10 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #3: c10d::ProcessGroupNCCL::WorkNCCL::isCompleted() + 0x68 (0x7f1150302828 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #4: c10d::ProcessGroupNCCL::Watchdog::runLoop() + 0x9c3 (0x7f11503071d3 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #5: c10d::ProcessGroupNCCL::Watchdog::run() + 0x157 (0x7f11503093b7 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #6: <unknown function> + 0xdc253 (0x7f120aeb0253 in /lib/x86_64-linux-gnu/libstdc++.so.6)
frame #7: <unknown function> + 0x94ac3 (0x7f120de6bac3 in /lib/x86_64-linux-gnu/libc.so.6)
frame #8: <unknown function> + 0x126850 (0x7f120defd850 in /lib/x86_64-linux-gnu/libc.so.6)

terminate called after throwing an instance of 'c10::DistBackendError'
  what():  [PG ID 0 PG GUID 0(default_pg) Rank 1] Process group watchdog thread terminated with exception: CUDA error: device-side assert triggered
Search for `cudaErrorAssert' in https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html for more information.
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

Exception raised from query at /pytorch/c10/cuda/CUDAEvent.h:111 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x9d (0x7f1204d4e05d in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libc10.so)
frame #1: <unknown function> + 0xc114 (0x7f1204de7114 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libc10_cuda.so)
frame #2: c10d::ProcessGroupNCCL::WorkNCCL::finishedGPUExecutionInternal() const + 0x50 (0x7f11502f3d10 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #3: c10d::ProcessGroupNCCL::WorkNCCL::isCompleted() + 0x68 (0x7f1150302828 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #4: c10d::ProcessGroupNCCL::Watchdog::runLoop() + 0x9c3 (0x7f11503071d3 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #5: c10d::ProcessGroupNCCL::Watchdog::run() + 0x157 (0x7f11503093b7 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #6: <unknown function> + 0xdc253 (0x7f120aeb0253 in /lib/x86_64-linux-gnu/libstdc++.so.6)
frame #7: <unknown function> + 0x94ac3 (0x7f120de6bac3 in /lib/x86_64-linux-gnu/libc.so.6)
frame #8: <unknown function> + 0x126850 (0x7f120defd850 in /lib/x86_64-linux-gnu/libc.so.6)

Exception raised from run at /pytorch/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:2125 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x9d (0x7f1204d4e05d in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libc10.so)
frame #1: <unknown function> + 0x68ef03 (0x7f114fa8ef03 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #2: <unknown function> + 0xdc253 (0x7f120aeb0253 in /lib/x86_64-linux-gnu/libstdc++.so.6)
frame #3: <unknown function> + 0x94ac3 (0x7f120de6bac3 in /lib/x86_64-linux-gnu/libc.so.6)
frame #4: <unknown function> + 0x126850 (0x7f120defd850 in /lib/x86_64-linux-gnu/libc.so.6)

[rank0]:[E327 02:53:46.532756449 ProcessGroupNCCL.cpp:2119] [PG ID 0 PG GUID 0(default_pg) Rank 0] Process group watchdog thread terminated with exception: CUDA error: device-side assert triggered
Search for `cudaErrorAssert' in https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html for more information.
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

Exception raised from query at /pytorch/c10/cuda/CUDAEvent.h:111 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x9d (0x7f826118b05d in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libc10.so)
frame #1: <unknown function> + 0xc114 (0x7f826452e114 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libc10_cuda.so)
frame #2: c10d::ProcessGroupNCCL::WorkNCCL::finishedGPUExecutionInternal() const + 0x50 (0x7f81ac6f3d10 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #3: c10d::ProcessGroupNCCL::WorkNCCL::isCompleted() + 0x68 (0x7f81ac702828 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #4: c10d::ProcessGroupNCCL::Watchdog::runLoop() + 0x9c3 (0x7f81ac7071d3 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #5: c10d::ProcessGroupNCCL::Watchdog::run() + 0x157 (0x7f81ac7093b7 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #6: <unknown function> + 0xdc253 (0x7f82670b0253 in /lib/x86_64-linux-gnu/libstdc++.so.6)
frame #7: <unknown function> + 0x94ac3 (0x7f826a26bac3 in /lib/x86_64-linux-gnu/libc.so.6)
frame #8: <unknown function> + 0x126850 (0x7f826a2fd850 in /lib/x86_64-linux-gnu/libc.so.6)

terminate called after throwing an instance of 'c10::DistBackendError'
  what():  [PG ID 0 PG GUID 0(default_pg) Rank 0] Process group watchdog thread terminated with exception: CUDA error: device-side assert triggered
Search for `cudaErrorAssert' in https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html for more information.
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

Exception raised from query at /pytorch/c10/cuda/CUDAEvent.h:111 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x9d (0x7f826118b05d in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libc10.so)
frame #1: <unknown function> + 0xc114 (0x7f826452e114 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libc10_cuda.so)
frame #2: c10d::ProcessGroupNCCL::WorkNCCL::finishedGPUExecutionInternal() const + 0x50 (0x7f81ac6f3d10 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #3: c10d::ProcessGroupNCCL::WorkNCCL::isCompleted() + 0x68 (0x7f81ac702828 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #4: c10d::ProcessGroupNCCL::Watchdog::runLoop() + 0x9c3 (0x7f81ac7071d3 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #5: c10d::ProcessGroupNCCL::Watchdog::run() + 0x157 (0x7f81ac7093b7 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #6: <unknown function> + 0xdc253 (0x7f82670b0253 in /lib/x86_64-linux-gnu/libstdc++.so.6)
frame #7: <unknown function> + 0x94ac3 (0x7f826a26bac3 in /lib/x86_64-linux-gnu/libc.so.6)
frame #8: <unknown function> + 0x126850 (0x7f826a2fd850 in /lib/x86_64-linux-gnu/libc.so.6)

Exception raised from run at /pytorch/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:2125 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x9d (0x7f826118b05d in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libc10.so)
frame #1: <unknown function> + 0x68ef03 (0x7f81abe8ef03 in /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/lib/libtorch_cuda.so)
frame #2: <unknown function> + 0xdc253 (0x7f82670b0253 in /lib/x86_64-linux-gnu/libstdc++.so.6)
frame #3: <unknown function> + 0x94ac3 (0x7f826a26bac3 in /lib/x86_64-linux-gnu/libc.so.6)
frame #4: <unknown function> + 0x126850 (0x7f826a2fd850 in /lib/x86_64-linux-gnu/libc.so.6)

W0327 02:58:48.037000 6888 torch/distributed/elastic/multiprocessing/api.py:1012] Sending process 6989 closing signal SIGTERM
E0327 02:58:48.102000 6888 torch/distributed/elastic/multiprocessing/api.py:986] failed (exitcode: -6) local_rank: 0 (pid: 6988) of binary: /__w/trl/trl/.venv/bin/python3
Traceback (most recent call last):
  File "/__w/trl/trl/.venv/bin/accelerate", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/__w/trl/trl/.venv/lib/python3.12/site-packages/accelerate/commands/accelerate_cli.py", line 50, in main
    args.func(args)
  File "/__w/trl/trl/.venv/lib/python3.12/site-packages/accelerate/commands/launch.py", line 1392, in launch_command
    multi_gpu_launcher(args)
  File "/__w/trl/trl/.venv/lib/python3.12/site-packages/accelerate/commands/launch.py", line 1023, in multi_gpu_launcher
    distrib_run.run(args)
  File "/__w/trl/trl/.venv/lib/python3.12/site-packages/torch/distributed/run.py", line 981, in run
    elastic_launch(
  File "/__w/trl/trl/.venv/lib/python3.12/site-packages/torch/distributed/launcher/api.py", line 170, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/__w/trl/trl/.venv/lib/python3.12/site-packages/torch/distributed/launcher/api.py", line 317, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
trl/scripts/rloo.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2026-03-27_02:58:48
  host      : aws-g5-12xlarge-cache-use1-public-80-mchlv-runner-t7bq-workflow
  rank      : 1 (local_rank: 1)
  exitcode  : -6 (pid: 6989)
  error_file: <N/A>
  traceback : Signal 6 (SIGABRT) received by PID 6989
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2026-03-27_02:58:48
  host      : aws-g5-12xlarge-cache-use1-public-80-mchlv-runner-t7bq-workflow
  rank      : 0 (local_rank: 0)
  exitcode  : -6 (pid: 6988)
  error_file: <N/A>
  traceback : Signal 6 (SIGABRT) received by PID 6988
============================================================
```

--- a/tests/distributed/test_distributed.py
+++ b/tests/distributed/test_distributed.py
@@ -234,7 +234,14 @@ class TestDistributed(
                 ),
             ),
             pytest.param("zero3", marks=pytest.mark.xfail(reason="ZeRO 3 is currently failing, see #4899")),
-            "fsdp2",
+            pytest.param(
+                "fsdp2",
+                marks=pytest.mark.xfail(
+                    Version(transformers.__version__) == Version("5.4.0"),
+                    reason="Upstream issue: NaN weights on non-rank-0 FSDP processes (see #5386 and transformers#45050)",
+                    strict=True,
+                ),
+            ),
         ],
     )
     def test_rloo(self, config, get_config_path):

